NR-298654: Adding ubuntu 24 support for logs recipe

nr-rkallempudi opened the pull request.

nr-rkallempudi commented:
Extending SuperAgent and Infra agent support for logs on following distributions:
1. Ubuntu 24.04
2. Suse 15.*

danybmx reviewed (changes requested):
Hi @nr-rkallempudi thanks a lot for the contribution! Although I left some comments this is so nice!

danybmx reviewed (dismissed):
👌🏻

nr-rkallempudi commented on recipes/newrelic/infrastructure/logs/linux-logs.yml:
```diff
@@ -57,7 +57,7 @@ installTargets:
     os: linux
     platformFamily: suse
-    platformVersion: "(15\\.[1-4]|15-SP[1-4])"
+    platformVersion: "(15)\\.*"
```
yes. It is added considering that common yaml configuration available at this time for both OTel and Fluent Bit. Plan is there to separate configs for Otel and Fluent Bit so managing would be lot easier.

nr-rkallempudi replied:
It is clarified in following conversations that above config is used by only fluent bit. so remove all(.*) versions match and added only supported ones upto 15.5

Nandu-pns commented on test/definitions/logging/suse155-logs.json:
```diff
@@ -0,0 +1,34 @@
+                "source_repository": "https://github.com/newrelic/open-install-library",
+                "deploy_script_path": "test/deploy/linux/newrelic-cli/install/roles",
+                "params": {
+                    "validate_output": "Logs Integration\\s+\\(installed\\)",
```
It looks you're not specifying the logs integration target to install, could you please check ? 

For reference, please have a look at this - https://github.com/newrelic/open-install-library/blob/main/test/definitions/logging/amazonlinux2-logs.json#L26

Nandu-pns commented on test/definitions/super-agent/suse/suse155-super-agent.json:
```diff
@@ -0,0 +1,35 @@
+{
```
Recipes changes are done in the super-agent/logs, hence the test definition configs needs to be created for super-agent logs and not for super-agent.

At the moment the changes are done here `test/definitions/super-agent/` instead these should be done at `test/definitions/super-agent/logs`

nr-rkallempudi replied:
updated PR

nr-rkallempudi commented:
Following are the effective changes done as part of the PR:
1. Extended Logs support for Ubuntu 24.04
2. Extended Logs support for Suse 15.5
3. Removed installTargets of Suse 15.5 for aarch64 as Fluent Bit support is not available
4. Updated test cases. Tests are not validated on local host. Requested Virtuous team to run the tests before approving the PR

Nandu-pns commented on test/definitions/super-agent/logs/suse155-super-agent.json:
```diff
@@ -0,0 +1,36 @@
+                "source_repository": "https://github.com/newrelic/open-install-library.git",
+                "deploy_script_path": "test/deploy/linux/newrelic-cli/install-recipe/roles",
+                "params": {
+                    "newrelic_cli_option": "-n logs-integration",
```
The `newrelic_cli_option` param needs to be updated to `"-n super-agent,logs-integration-super-agent",` and "env_var": "NEW_RELIC_CLI_SKIP_CORE=1" param needs to be added.

An example test config with such params here https://github.com/newrelic/open-install-library/blob/main/test/definitions/super-agent/logs/suse154-super-agent-logs.json#L28-L29

Nandu-pns commented on test/definitions/super-agent/logs/suse155-super-agent.json:
```diff
@@ -0,0 +1,36 @@
+                "params": {
+                    "newrelic_cli_option": "-n logs-integration",
+                    "recipe_content_url": "https://raw.githubusercontent.com/newrelic/open-install-library/main/recipes/newrelic/infrastructure/super-agent/suse.yml",
+                    "validate_output": "New Relic Super Agent\\s+\\(installed\\)",
```
`validate_output` should be updated to validate logs integration as below.

`"validate_output": "Logs Integration\\s+\\(installed\\)",`

Nandu-pns commented:
Hi @nr-rkallempudi - Changes looks good to us. I was able to test them out and could see `logs-integration` is being installed successfully on the latest host versions.

nr-rkallempudi commented:
Done validations on Suse 15.4 and Suse 15.5 and identified that Fluent bit is not installing on all AMIs of SLES 15.4 and 15.5. 

Created Jira task to track this one on priority.
https://new-relic.atlassian.net/browse/NR-301797

**Screenshots of successful installation of fluent bit on EC2 hosts:**
<img width="1032" alt="Screenshot 2024-08-10 at 2 45 23 PM" src="https://github.com/user-attachments/assets/25e0eeab-0f3b-4b82-bfa9-938a5faf1501">

<img width="931" alt="Screenshot 2024-08-10 at 2 46 02 PM" src="https://github.com/user-attachments/assets/dae1d4cf-30d9-4762-8615-237dd9317453">

Avinash-Sanpala commented on test/definitions/super-agent/logs/suse155-super-agent-logs.json:
```diff
@@ -0,0 +1,37 @@
+            "provider": "aws",
+            "type": "ec2",
+            "size": "t3.nano",
+            "ami_name": "suse-sles-15-sp5-chost-byos-v????????-hvm-ssd-x86_64"
```
@nr-rkallempudi just want to make sure if this AMI is compatible with the ami instance mentioned in the fluent bit packages of the respective version? Also, If you think this has to be modified upon checking in the fluent bit packages repo, please modify the instance type accordingly similar to the changes done to above logs deploy config files

Avinash-Sanpala reviewed (approved):
LGTM!
